iot-e2e-tests/common/src/test/java/tests/integration/com/microsoft/azure/sdk/iot/helpers/IotHubServicesCommon.java

```diff
@@ -55,12 +55,16 @@ public static void sendMessages(InternalClient client,
 
             for (int i = 0; i < messagesToSend.size(); ++i)
             {
-                if (isErrorInjectionMessage(messagesToSend.get(i)))
+                if ((protocol == IotHubClientProtocol.MQTT || protocol == IotHubClientProtocol.MQTT_WS) && isErrorInjectionMessage(messagesToSend.get(i)))
```

Comment:
Q - why do we need these changes here, and in `sendErrorInjectionMessageAndWaitForResponse` both?

Reply:
I don't think there is a good reason. There's some logic here that could certainly be centralized. I'll take that in a separate PR though